src/redisjson.rs

```diff
@@ -81,8 +88,11 @@ impl RedisJSON {
         Ok(s.to_string())
     }
 
-    pub fn get_doc(&self, path: &str) -> Result<Value, Error> {
-        let results = jsonpath::select(&self.data, path)?;
-        Ok(results)
+    fn get_doc<'a>(&'a self, path: &'a str) -> Result<&'a Value, Error> {
```

Comment:
Why is the explicit lifetime annotation needed here? Seems unnecessary.

Reply:
```
error[E0623]: lifetime mismatch
  --> src/redisjson.rs:94:24
   |
91 |     fn get_doc(&self, path: &str) -> Result<&Value, Error> {
   |                             ----     ---------------------
   |                             |
   |                             this parameter and the return type are declared with different lifetimes...
...
94 |             Some(s) => Ok(s),
   |                        ^^^^^ ...but data from `path` is returned here
```

Reply:
You're right, now that I've looked at it again it makes perfect sense, since you're borrowing from both `self` and `path`.